Userland/Libraries/LibJS/SourceTextModule.h

```diff
@@ -26,7 +26,7 @@ class SourceTextModule final : public CyclicModule {
     virtual ThrowCompletionOr<ResolvedBinding> resolve_export(VM& vm, DeprecatedFlyString const& export_name, Vector<ResolvedBinding> resolve_set = {}) override;
 
     Object* import_meta() { return m_import_meta; }
-    void set_import_meta(Badge<MetaProperty>, Object* import_meta) { m_import_meta = import_meta; }
+    void set_import_meta(Object* import_meta) { m_import_meta = import_meta; }
```

Comment:
This is probably okay, but to keep the restriction around, you can:
```suggestion
    template<OneOf<MetaProperty, Bytecode::GetImportMeta> T>
    void set_import_meta(Badge<T>, Object* import_meta) { m_import_meta = import_meta; }
```

Reply:
after moving the implementation to vm I am not sure if this is possible anymore, but I could definitely be wrong

Reply:
if the only thing accessing this is VM, you can take a `Badge<VM>` similar to how this was originally.